### PR TITLE
Implemented /time set x

### PR DESCRIPTION
**Pull request by robinkanters**
User can now use `/time set x` to set the time. The time is then synced to the clients

### CODE REVIEW

**Comment by SirCmpwn** [comment]
Thanks @robinkanters! Before I review this, could you please [sign the contributor agreement](https://www.clahub.com/agreements/SirCmpwn/TrueCraft)?

**Comment by robinkanters** [comment]
@SirCmpwn getting an error message when submitting the form

**Comment by SirCmpwn** [comment]
Can you share the error message with me?

**Comment by robinkanters** [comment]
(pasted screenie above, that's all I see)

**Comment by SirCmpwn** [comment]
Thanks. I'll direct CLAHub to this thread. In the meanwhile, can you email your agreement to sir@cmpwn.com and I'll keep a record of it myself?

**Comment by robinkanters** [comment]
Just a screenshot of the form, or what?

**Comment by SirCmpwn** [comment]
That works for me. It looks like CLAHub is dead (at least according to the Github issues), so I'll have to work out some other solution for having people sign the agreement.

**Comment by robinkanters** [comment]
Sent

**Comment by SirCmpwn** [comment]
Thanks! Sorry for the trouble.

File: TrueCraft/Commands/DebugCommands.cs [hunk 1]
```diff
@@ -94,12 +94,30 @@ public override string[] Aliases
...
+
+                    int newTime;
+
+                    if(!Int32.TryParse(Arguments[2], out newTime))
+                        Help(Client, Alias, Arguments);
+
+                    Client.World.Time = newTime;
+
+                    Client.SendMessage(string.Format("Setting time to {0}", Arguments[2]));
+                    Client.QueuePacket(new TimeUpdatePacket(newTime));
```
Comment (SirCmpwn) [inline TrueCraft/Commands/DebugCommands.cs]: You'll want to send this packet to all clients in that world, not just the one issuing the command.

**Comment by SirCmpwn** [comment]
Looks good minus one minor issue.

**Comment by robinkanters** [comment]
Hang on

**Comment by robinkanters** [comment]
That should do it!

**Comment by SirCmpwn** [comment]
Almost there! TrueCraft is being built to support multiple worlds. You can't just send a time update to all clients - check that `client.World` is equal to the world that the time is being set in.

**Comment by robinkanters** [comment]
The chat system is currently inter-world, too... Is that intentional, then?

**Comment by SirCmpwn** [comment]
Yes. You can chat with the rest of the server from the nether, for example (the nether being implemented as a separate world).

**Comment by SirCmpwn** [comment]
Thanks for the PR!

**Comment by robinkanters** [comment]
Okay then :+1:

**Pull request merged**